### ISSUE TITLE
73 tests fail with TypeError in test/run-spec-wasm2c.py:101

### ISSUE THREAD

**Issue by rathann** [WebAssembly/wabt#1181]
In 1.0.12, after removing test/regress/regress-30.txt to work around #1180 , 73 tests fail with this traceback:
```
- test/wasm2c/spec/address.txt                                                                      
  expected error code 0, got 1.
  STDERR MISMATCH:
  --- expected
  +++ actual
  @@ -0,0 +1,12 @@
  +Traceback (most recent call last):
  +  File "/builddir/build/BUILD/wabt-1.0.12/test/run-spec-wasm2c.py", line 418, in <module>
  +    sys.exit(main(sys.argv[1:]))
  +  File "/builddir/build/BUILD/wabt-1.0.12/test/run-spec-wasm2c.py", line 383, in main
  +    cwriter.Write()
  +  File "/builddir/build/BUILD/wabt-1.0.12/test/run-spec-wasm2c.py", line 127, in Write
  +    self._CacheModulePrefixes()
  +  File "/builddir/build/BUILD/wabt-1.0.12/test/run-spec-wasm2c.py", line 150, in _CacheModulePrefixes
  +    name = MangleName(name)
  +  File "/builddir/build/BUILD/wabt-1.0.12/test/run-spec-wasm2c.py", line 101, in MangleName
  +    if c in '_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXY0123456789':
  +TypeError: 'in <string>' requires string as left operand, not int
  STDOUT MISMATCH:
  --- expected
  +++ actual
  @@ -1 +0,0 @@
  -238/238 tests passed.
```

Failing tests:
```
- test/wasm2c/spec/address.txt
- test/wasm2c/spec/align.txt
- test/wasm2c/spec/binary.txt
- test/wasm2c/spec/block.txt
- test/wasm2c/spec/br.txt
- test/wasm2c/spec/br_if.txt
- test/wasm2c/spec/br_table.txt
- test/wasm2c/spec/break-drop.txt
- test/wasm2c/spec/call.txt
- test/wasm2c/spec/call_indirect.txt
- test/wasm2c/spec/comments.txt
- test/wasm2c/spec/const.txt
- test/wasm2c/spec/conversions.txt
- test/wasm2c/spec/custom.txt
- test/wasm2c/spec/data.txt
- test/wasm2c/spec/elem.txt
- test/wasm2c/spec/endianness.txt
- test/wasm2c/spec/exports.txt
- test/wasm2c/spec/f32.txt
- test/wasm2c/spec/f32_bitwise.txt
- test/wasm2c/spec/f32_cmp.txt
- test/wasm2c/spec/f64.txt
- test/wasm2c/spec/f64_bitwise.txt
- test/wasm2c/spec/f64_cmp.txt
- test/wasm2c/spec/fac.txt
- test/wasm2c/spec/float_exprs.txt
- test/wasm2c/spec/float_literals.txt
- test/wasm2c/spec/float_memory.txt
- test/wasm2c/spec/float_misc.txt
- test/wasm2c/spec/forward.txt
- test/wasm2c/spec/func.txt
- test/wasm2c/spec/func_ptrs.txt
- test/wasm2c/spec/globals.txt
- test/wasm2c/spec/i32.txt
- test/wasm2c/spec/i64.txt
- test/wasm2c/spec/if.txt
- test/wasm2c/spec/imports.txt
- test/wasm2c/spec/inline-module.txt
- test/wasm2c/spec/int_exprs.txt
- test/wasm2c/spec/int_literals.txt
- test/wasm2c/spec/labels.txt
- test/wasm2c/spec/left-to-right.txt
- test/wasm2c/spec/linking.txt
- test/wasm2c/spec/load.txt
- test/wasm2c/spec/local_get.txt
- test/wasm2c/spec/local_set.txt
- test/wasm2c/spec/local_tee.txt
- test/wasm2c/spec/loop.txt
- test/wasm2c/spec/memory.txt
- test/wasm2c/spec/memory_grow.txt
- test/wasm2c/spec/memory_redundancy.txt
- test/wasm2c/spec/memory_size.txt
- test/wasm2c/spec/memory_trap.txt
- test/wasm2c/spec/names.txt
- test/wasm2c/spec/nop.txt
- test/wasm2c/spec/return.txt
- test/wasm2c/spec/select.txt
- test/wasm2c/spec/skip-stack-guard-page.txt
- test/wasm2c/spec/stack.txt
- test/wasm2c/spec/start.txt
- test/wasm2c/spec/store.txt
- test/wasm2c/spec/switch.txt
- test/wasm2c/spec/token.txt
- test/wasm2c/spec/traps.txt
- test/wasm2c/spec/type.txt
- test/wasm2c/spec/typecheck.txt
- test/wasm2c/spec/unreachable.txt
- test/wasm2c/spec/unwind.txt
- test/wasm2c/spec/unreached-invalid.txt
- test/wasm2c/spec/utf8-custom-section-id.txt
- test/wasm2c/spec/utf8-import-field.txt
- test/wasm2c/spec/utf8-import-module.txt
- test/wasm2c/spec/utf8-invalid-encoding.txt
```

**Comment by rathann** [comment]
These tests pass with python2, so again, this seems to be python3-specific issue like #1180 .

**Comment by rathann** [comment]
In python2, `s.encode('utf-8')` at line 99 of [`test/run-spec-wasm2c.py`](https://github.com/WebAssembly/wabt/blob/master/test/run-spec-wasm2c.py#L99) returns a `'str'` type in python2 while it returns class `'bytes'` in python3. See [Text versus binary data](https://docs.python.org/3/howto/pyporting.html#text-versus-binary-data).

**Comment by rathann** [comment]
Quick and dirty fix that works in both python2 and python3:
``` diff
--- wabt-1.0.12/test/run-spec-wasm2c.py.orig    2019-09-25 23:04:14.000000000 +0200
+++ wabt-1.0.12/test/run-spec-wasm2c.py 2019-10-09 10:14:41.905966919 +0200
@@ -98,6 +98,8 @@ def MangleName(s):
     result = 'Z_'
     for c in s.encode('utf-8'):
         # NOTE(binji): Z is not allowed.
+        if type(c) == int:
+            c=chr(c)
         if c in '_abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXY0123456789':
             result += c
         else:
```

**Comment by binji** [comment]
I think this should be fixed with #1285. Please re-open if not.
